### ISSUE TITLE
IPv6 Firewall rules always recreated

### ISSUE THREAD

**Issue by jdmulloy** [squat/terraform-provider-vultr#10]
Hello,

Thanks for making the provider. I'm working on setting up Terraform to manage my Vultr resources. I'm setting up a new firewall group since I can't import my existing group and editing testate isn't worth the effort. I'm having an issue with my IPv6 rules to allow access from `::/0`. Terraform always wants to delete and recreate them, because it's seeing the cider_block as being `0.0.0.0/0` instead of `::/0`. It creates the rules just fine, it just doesn't believe that they are setup.

# Terraform DSL
```
resource "vultr_firewall_group" "default" {
  description = "Terraform managed firewall group"
}

resource "vultr_firewall_rule" "ipv6_test" {
  firewall_group_id = "${vultr_firewall_group.default.id}"
  cidr_block = "::/0"
  protocol = "tcp"
  from_port = "444"
  to_port = "444"
}
```

# Terraform plan output
```
root@server1:/data/terraform/jdmulloy-vultr-terraform # terraform plan --target vultr_firewall_rule.ipv6_test                                                                                          
Refreshing Terraform state in-memory prior to plan...                                              
The refreshed state will be used to calculate this plan, but will not be                           
persisted to local or remote state storage.      

vultr_firewall_group.default: Refreshing state... (ID: c695ecf4)                                   
vultr_firewall_rule.ipv6_test: Refreshing state... (ID: c695ecf4/33)                               

------------------------------------------------------------------------                           

An execution plan has been generated and is shown below.                                           
Resource actions are indicated with the following symbols:                                         
-/+ destroy and then create replacement          

Terraform will perform the following actions:    

-/+ vultr_firewall_rule.ipv6_test (new resource required)                                          
      id:                "c695ecf4/33" => <computed> (forces new resource)                         
      action:            "accept" => <computed>  
      cidr_block:        "0.0.0.0/0" => "::/0" (forces new resource)                               
      direction:         "in" => <computed>      
      firewall_group_id: "c695ecf4" => "c695ecf4"                                                  
      from_port:         "444" => "444"          
      protocol:          "tcp" => "tcp"          
      to_port:           "444" => "444"          


Plan: 1 to add, 0 to change, 1 to destroy.
```

**Comment by squat** [comment]
@jdmulloy thanks for finding this. Turns out it is an faulty assumption in the firewall rule unmarshalling logic in the Vultr API library. I put up a PR to fix it [0]. Once that lands, I'll update this project's dependencies to fix this issue.

[0] https://github.com/JamesClonk/vultr/pull/48

**Comment by jdmulloy** [comment]
Awesome! Thanks.

As a work around I've set the value on all my IPv6 rules to "0.0.0.0/0" so that Terraform won't try to reconfigure them.

**Comment by squat** [comment]
@jdmulloy the upstream issue was merged and I created a new commit that fixes this issue: 82c5e9ddd85bf72be86c6c5f957f3e3dbe77d3fb. Please re-open this if you keep seeing these issues.
